[CS1998] This async method lacks 'await' operators and will run synchronously

eaba opened the pull request:
## Changes

"warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread."

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [ ] This change follows the [Akka.NET API Compatibility Guidelines](https://getakka.net/community/contributing/api-changes-compatibility.html).
* [ ] This change follows the [Akka.NET Wire Compatibility Guidelines](https://getakka.net/community/contributing/wire-compatibility.html).
* [ ] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed, if any.
* [ ] I have added website documentation for this feature.

### Latest `dev` Benchmarks 

Include data from the [relevant benchmark](https://getakka.net/community/contributing/index.html#improve-performance) prior to this change here.

### This PR's Benchmarks

Include data from after this change here.

eaba committed:
This async method lacks 'await' operators and will run synchronously

eaba commented on src/core/Akka.Persistence/Journal/MemoryJournal.cs:
```diff
@@ -184,7 +184,8 @@ protected override bool ReceivePluginInternal(object message)
         private async Task<(IEnumerable<string> Ids, int LastOrdering)> SelectAllPersistenceIdsAsync(int offset)
         {
-            return (new HashSet<string>(_allMessages.Skip(offset).Select(p => p.PersistenceId)), _allMessages.Count); 
+            return await Task.FromResult((new HashSet<string>(_allMessages.Skip(offset).Select(p => p.PersistenceId)), _allMessages.Count));
```
Let's talk about this @Aaronontheweb @Arkatufus  

![image](https://user-images.githubusercontent.com/8049896/224715005-54c3295e-d070-465e-a988-9c12e22bbcf0.png)

Aaronontheweb replied:
I'd remove the `async` and the `await` here - just returning the `Task.Result` should be fine.

eaba commented on src/core/Akka.Persistence/Journal/MemoryJournal.cs:
```diff
@@ -207,7 +208,8 @@ private async Task<int> ReplayTaggedMessagesAsync(ReplayTaggedMessages replay)
             }
 
-            return _tagsToMessagesMapping[replay.Tag].Count - 1;
+            return await Task.FromResult(_tagsToMessagesMapping[replay.Tag].Count - 1);
```
Let's talk about this @Aaronontheweb @Arkatufus 

![image](https://user-images.githubusercontent.com/8049896/224714853-2b77732c-2664-47a9-b0b5-141aa7f66e01.png)

Aaronontheweb replied:
Same feedback as above.

eaba committed:
removed `async`/`await`

eaba committed:
Merge branch 'dev' into CS1998

Aaronontheweb commented:
@eaba is this ready for review?

eaba commented:
I will tell you @Aaronontheweb

eaba committed:
[improves] - `AsyncEnumerable.cs`
- `TestPublisher.cs`
- `Utils.cs`
- `EventFilterApplier.cs`
- `BlockingQueue.cs`
- `TestKitBase_AwaitConditions.cs`
- `TestKitBase_Within.cs`

eaba committed:
Fix `CS1998`

Aaronontheweb commented:
```
Akka.Tests.Actor.LocalActorRefProviderSpec.An_ActorRefFactory_must_only_create_one_instance_of_an_actor_with_a_specific_address_in_a_concurrent_environment
Akka.Actor.InvalidActorNameException : Actor name "new-actor0" is not unique!
```

This is broken

eaba committed:
changed it `Task.FromResult` to `Task.Run`
